libs/ui-components/src/components/form/validations.ts

```diff
@@ -4,7 +4,26 @@ import { FlightCtlLabel } from '../../types/extraTypes';
 
 type UnvalidatedLabel = Partial<FlightCtlLabel>;
 
-export const uniqueLabelKeysSchema = (t: TFunction) =>
+const SYSTEMD_PATTERNS_REGEXP = /^[a-z][a-z0-9-_.]*$/;
+const SYSTEMD_UNITS_MAX_PATTERNS = 256;
+
+const K8S_LABEL_REGEXP = /^[a-z0-9]([-a-z0-9]*[a-z0-9])?$/;
```

Comment:
This regex is incorrect IMO.

we need to split it as different parts of label have different requirements
key - `([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]`
key prefix - `[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`
value - (([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?

also lengths of each part needs to be verified as described in https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

Reply:
Agree. As before, since this requires additional checks and UX, I can handle that separately.